Deprecated host_key_checking option

Akasurde opened the pull request:
##### SUMMARY

* User can set host_key_checking configuration from
  connection plugins such as ssh and paramiko_ssh itself.
  Deprecate host_key_checking from global configuration.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

bcoca commented:
guessing https://github.com/ansible/ansible/pull/81947 would fix the issue of just 'constant' access vs specific setting

Akasurde commented:
/azp run

Akasurde committed:
Deprecated host_key_checking option

* User can set host_key_checking configuration from
  connection plugins such as ssh and paramiko_ssh itself.
  Deprecate host_key_checking from global configuration.

Signed-off-by: Abhijeet Kasurde <Akasurde@redhat.com>
